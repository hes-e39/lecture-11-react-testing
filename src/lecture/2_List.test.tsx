import { describe, expect, it } from 'vitest';
import { render } from 'vitest-browser-react';

import List from './2_List';

describe('A List', () => {
    it('should render custom title', async () => {
        const { getByText } = render(<List title="My custom list" />);

        await expect.element(getByText('My custom list')).toBeInTheDocument();
    });

    it('should render list items', async () => {
        const data = ['item 1', 'item 2', 'item 3'];
        const { getByText } = render(<List items={data} />);

        await expect.element(getByText('item 1')).toBeInTheDocument();
        await expect.element(getByText('item 2')).toBeInTheDocument();
        await expect.element(getByText('item 3')).toBeInTheDocument();
    });

    // snapshot tests
    it('should render list (snapshot)', () => {
        const result = render(<List title="My custom list" />);
        expect(result).toMatchSnapshot();

        // // update with some different props
        const result2 = render(<List items={['item 1', 'item 2', 'item 3']} />);
        expect(result2).toMatchSnapshot();
    });
});
