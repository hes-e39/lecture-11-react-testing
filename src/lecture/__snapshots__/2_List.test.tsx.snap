// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`A List > should render list (snapshot) 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    
    
    
  


    <vite-plugin-checker-error-overlay />
    <div>
      <div
        class="border border-gray-300 p-4 rounded-lg w-fit"
      >
        <h3
          class="text-lg font-bold"
        >
          My custom list
        </h3>
        <ul
          class=""
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="border border-gray-300 p-4 rounded-lg w-fit"
    >
      <h3
        class="text-lg font-bold"
      >
        My custom list
      </h3>
      <ul
        class=""
      />
    </div>
  </div>,
  "debug": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholder": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`A List > should render list (snapshot) 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    
    
    
  


    <vite-plugin-checker-error-overlay />
    <div>
      <div
        class="border border-gray-300 p-4 rounded-lg w-fit"
      >
        <h3
          class="text-lg font-bold"
        >
          My custom list
        </h3>
        <ul
          class=""
        />
      </div>
    </div>
    <div>
      <div
        class="border border-gray-300 p-4 rounded-lg w-fit"
      >
        <h3
          class="text-lg font-bold"
        >
          My List
        </h3>
        <ul
          class=""
        >
          <li
            class="p-2 bg-gray-100 rounded-lg my-2 w-[150px] text-gray-600 flex gap-2 truncate"
          >
            <div
              class="font-bold"
            >
              1
              .
            </div>
            <div
              class=""
            >
              item 1
            </div>
          </li>
          <li
            class="p-2 bg-gray-100 rounded-lg my-2 w-[150px] text-gray-600 flex gap-2 truncate"
          >
            <div
              class="font-bold"
            >
              2
              .
            </div>
            <div
              class=""
            >
              item 2
            </div>
          </li>
          <li
            class="p-2 bg-gray-100 rounded-lg my-2 w-[150px] text-gray-600 flex gap-2 truncate"
          >
            <div
              class="font-bold"
            >
              3
              .
            </div>
            <div
              class=""
            >
              item 3
            </div>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="border border-gray-300 p-4 rounded-lg w-fit"
    >
      <h3
        class="text-lg font-bold"
      >
        My List
      </h3>
      <ul
        class=""
      >
        <li
          class="p-2 bg-gray-100 rounded-lg my-2 w-[150px] text-gray-600 flex gap-2 truncate"
        >
          <div
            class="font-bold"
          >
            1
            .
          </div>
          <div
            class=""
          >
            item 1
          </div>
        </li>
        <li
          class="p-2 bg-gray-100 rounded-lg my-2 w-[150px] text-gray-600 flex gap-2 truncate"
        >
          <div
            class="font-bold"
          >
            2
            .
          </div>
          <div
            class=""
          >
            item 2
          </div>
        </li>
        <li
          class="p-2 bg-gray-100 rounded-lg my-2 w-[150px] text-gray-600 flex gap-2 truncate"
        >
          <div
            class="font-bold"
          >
            3
            .
          </div>
          <div
            class=""
          >
            item 3
          </div>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholder": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
